docs/examples/encrypted-events.mdx

```diff
@@ -0,0 +1,285 @@
+[Pattern 2](#pattern-2-deriving-a-symmetric-key-ecdh), and
+[Pattern 3](#pattern-3-on-chain-key-generation-rofl-friendly).
+
+### The `Encrypted` Event
+
+Define a standard event signature in your contract. This structure is shared by all patterns.
+
+```solidity
+// In your contract
+event Encrypted(address indexed sender, bytes32 nonce, bytes ciphertext);
+```
+
+- `sender`: The address that initiated the encryption. Indexing it helps in filtering events.
+- `nonce`: A unique number used once per encryption with a given key. **Crucially, never reuse a key/nonce pair.**
+- `ciphertext`: The encrypted data payload.
+
+### On-Chain Encryption
+
+Sapphire provides a precompiled contract for encryption. The basic flow inside a contract function is:
+
+1. **Get a Nonce:** Generate a fresh, random nonce for each encryption. Using a domain separator is a good practice.
+
+   ```solidity
+   // A 32-byte nonce for Deoxys-II, which uses the first 15 bytes
+   bytes32 nonce = bytes32(Sapphire.randomBytes(32, bytes("my-dapp-nonce")));
+   ```
+
+2. **Encrypt the Data:** Use `Sapphire.encrypt` with a key, the nonce, your **text** (plaintext), and optional authenticated data (**AAD**, named `ad` in the API).
+
+   ```solidity
+   // Keep names identical to the library
+   bytes memory ad = "additional data";
+   bytes memory encrypted = Sapphire.encrypt(key, nonce, text, ad);
+   ```
+
+3. **Emit the Event:**
+
+   ```solidity
+   emit Encrypted(msg.sender, nonce, encrypted);
+   ```
+
+### Off-Chain Decryption
+
+To read the secret text, an off-chain client with the correct key listens for `Encrypted` events and decrypts the `ciphertext`.
+
+```typescript
+import { AEAD, NonceSize } from '@oasisprotocol/deoxysii';
+import { ethers } from 'ethers';
+
+// You need:
+// - key: A Uint8Array(32) symmetric key
+// - nonceFromEvent: The 'nonce' field from the event log
+// - ciphertextFromEvent: The 'ciphertext' field from the event log
+// - adBytes: AAD bytes if used on-chain (a.k.a. "authenticated data")
+
+const aead = new AEAD(key);
+
+const plaintext = aead.decrypt(
+  // IMPORTANT: Deoxys-II uses a 15-byte nonce.
+  // We slice the first 15 bytes from the 32-byte value stored on-chain.
+  ethers.getBytes(nonceFromEvent).slice(0, NonceSize),
+  ethers.getBytes(ciphertextFromEvent),
+  adBytes,
+);
+
+console.log('Decrypted message:', ethers.toUtf8String(plaintext));
+```
+
+---
+
+## Pattern 1: Passing a Symmetric Key
+
+This is the simplest approach. The client generates a 32‑byte symmetric key and passes it as an argument in the (Sapphire‑encrypted) transaction to the smart contract.
+
+### TypeScript: generate a `bytes32` key
+
+```ts
+import { ethers } from 'ethers';
+import { randomBytes } from 'crypto';
+
+// Option A (ethers v6)
+const keyHex = ethers.hexlify(ethers.randomBytes(32)) as `0x${string}`;
+
+// Option B (Node.js crypto)
+const keyHex2 = ('0x' + randomBytes(32).toString('hex')) as `0x${string}`;
```

Comment:
Is `0x` redundant here?

Reply:
ethers v6 expects hex strings to be 0x prefixed. replaced with `ethers.hexlify`